leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceService.java

```diff
@@ -39,12 +40,34 @@
      * @param listener target to be removed.
      */
     void removeListener(PresenceListener listener);
-    
+
     /**
      * Returns the current status of a given LWM2M client registration.
      * 
      * @param registration data to check
-     * @return true if the status is {@link Presence#ONLINE}
+     * @return true if the status is {@link Presence#SLEEPING}
+     */
+    boolean isClientSleeping(Registration registration);
+
+    /**
+     * Set the state of the client identified by registration as {@link Presence#AWAKE}
+     * 
+     * @param registration the client's registration object
+     */
+    public void setAwake(Registration registration);
+
+    /**
+     * Notify the listeners that the client state changed to {@link Presence#SLEEPING}. The state changes is produced
+     * inside {@link LwM2mQueue} when the timer expires or when the client doesn't respond to a request.
+     * 
+     * @param registration the client's registration object
+     */
+    public void notifySleeping(Registration registration);
+
+    /**
+     * Creates a new {@link LwM2mQueue} object associated with the client.
+     * 
+     * @param registration the client's registration object.
      */
-    boolean isOnline(Registration registration);
+    public void createQueueObject(Registration registration);

```

Comment:
I think this is internal API and we don't want to expose it.

Reply:
Do you mean these methods shouldn't be public, or that they shouldn't be part of the interface at all?